src/vnm_mad/remotes/vxlan/vxlan.rb

```diff
@@ -67,20 +69,20 @@ def create_vlan_dev
 
         # `ip link add ...` returns 2 when vxlan device already exists
         # allow it to prevent race conditions
-        OpenNebula.exec_and_log("#{command(:ip)} link add #{@nic[@attr_vlan_dev]}"\
+        LocalCommand.run_sh("#{command(:ip)} link add #{@nic[@attr_vlan_dev]}"\
             " #{mtu} type vxlan id #{@nic[@attr_vlan_id]} #{group} #{ttl}"\
-            " #{tep} #{ip_link_conf}", nil, 2)
+            " #{tep} #{ip_link_conf}", :ok_rcs => 2)
 
-        OpenNebula.exec_and_log("#{command(:ip)} link set #{@nic[@attr_vlan_dev]} up")
+        LocalCommand.run_sh("#{command(:ip)} link set #{@nic[@attr_vlan_dev]} up")
     end
 
     def delete_vlan_dev
-        OpenNebula.exec_and_log("#{command(:ip)} link delete #{@nic[@attr_vlan_dev]}")
+        LocalCommand.run_sh("#{command(:ip)} link delete #{@nic[@attr_vlan_dev]}")
     end
 
     def list_interface_vlan(name)
-        text = %x(#{command(:ip_unpriv)} -d link show #{name})
-        return nil if $?.exitstatus != 0
+        text = `#{command(:ip_unpriv)} -d link show #{name}`
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Detected a dangerous subshell execution vulnerability where user input is directly used in a shell command without proper sanitisation. This could allow an attacker to inject and execute arbitrary system commands, potentially leading to system compromise or data breach. To remediate, avoid using shell commands with user input when possible, or implement strict input validation and use shell escaping functions like Shellwords.escape(). If shell execution is necessary, consider using a allow list of allowed commands and arguments. Link to more info : https://guides.rubyonrails.org/security.html